### PR TITLE
CCM-11214-Perspective Preferences

### PR DESCRIPTION
**Pull request by charanya-j**
Technically reviewed PR: https://github.com/harness/developer-hub/pull/2549

### CODE REVIEW

File: docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+#### AWS discount types supported by Harness CCM
+Currently, Harness CCM considers only three types of discounts under the LineItemType:
+
+* **BundledDiscount**: A usage-based discount that provides free or discounted usage of a service or feature based on the usage of another service or feature.
+
+* **PrivateRateDiscount**: Discount provided by AWS for using private pricing rates. When customers negotiate custom pricing with AWS, they may receive discounted rates for specific services based on their agreement.
+
+* **EdpDiscount**: Discounts associated with the use of AWS Enterprise Discount Program (EDP). The AWS Enterprise Discount Program offers volume-based discounts to customers who commit to certain usage levels and financial commitments with AWS.
+
+<!--The table below provides details on the AWS `LineItemType` that is associated or utilized with each AWS cost type:
```
Comment (sdsingh07) [inline docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md:109]: Is it ok to remain the commented code?

File: docs/cloud-cost-management/3-use-ccm-cost-reporting/3-root-cost-analysis/analyze-cost-for-gcp-using-perspectives.md [hunk 2]
```diff
@@ -48,18 +51,38 @@ The **Perspectives** page provides deep insights into your GCP costs. The cost i
...
+3. Select **GCP**.
+4. Select from the following filter options. You can add more than one filter.
+
+  - **Invoice Month**: Specific billing time frame during which usage of GCP resources and services is recorded, and the corresponding charges are calculated.
+  - **Product**: The various cloud services and offerings provided by GCP. 
+  - **Project**: A fundamental organizational unit that allows you to manage and organize your cloud resources.
+  - **SKUs**: Stock Keeping Unit (SKU) is a unique identifier used to represent specific types or configurations of cloud resources and services offered by Google. Each SKU corresponds to a particular product, service plan, or pricing option available in GCP.
+
+:::important note
+   - If you select the **Invoice Month** filter, the time filter option in the perspective is disabled.
```
Comment (sdsingh07) [inline docs/cloud-cost-management/3-use-ccm-cost-reporting/3-root-cost-analysis/analyze-cost-for-gcp-using-perspectives.md:69]: We can add the image when we hover on the time filter if GCP invoice month is selected.

Image speaks more than words :p

File: docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+You can configure perspective preferences in the **Default Settings** page under **Setup**. You can override these settings at the perspective level if needed. The following table provides the default values of perspective preferences:  
+
+| **General** | **AWS** | **GCP** |
+| --- | --- | --- |
+| Show Anomalies - Enabled | Include Discounts - Enabled | Include Discounts - Disabled |
+| Show Others - Disabled | Include Credit - Enabled | Include Taxes - Enabled |
+| Show Unallocated cost in Clusters - Disabled | Include Refunds - Enabled | - |
+| - | Include Taxes - Enabled | - |
+| - | Show costs as:<ul><li>Amortised</li><li>Net-amortised</li><li>Unblended</li><li>Blended</li><li>Effective</li></ul>Default value: Unblended | - |
+
```
Comment (sdsingh07) [inline docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md:19]: Also, need to mention that for all the existing perspectives where Show Others and Show Unallocated cost in Clusters are true, they will remain as true.

File: docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md (hunk 1 above)
Comment (charanya-j) [reply docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md:109]: yeah, that's not a problem. It won't show in the output.

File: docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md (hunk 3 above)
Comment (charanya-j) [reply docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md:19]: Isn't this just for the ones that are getting migrated? This won't be a use case ..say after a couple of months, correct? If I am correct, I don't think it's appropriate to add this in the core doc. I can add it to the release notes.

File: docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+| **AWS Cost Type**  | **Discount** | **Refund** | **Tax** | **Credit** |
+| --- | --- | --- | --- | --- |
+-->
+
+To know how to analyze the AWS perspective data, go to [Analyze cost for AWS](../3-root-cost-analysis/analyze-cost-for-aws.md).
+## GCP preferences
+The following cost factors retrieved from your GCP Billing Export data can be included or excluded in your perspectives for a more comprehensive view of your cloud costs.
+* **Include Discounts**: Includes any discounts that GCP applied to your usage. 
+* **Include Taxes**: Includes any taxes that GCP applied to your bill. For example, VAT or US sales tax. This option is enabled by default.
```
[the remark this answers is not in the document]
Comment (charanya-j) [reply docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md:119]: I've already provided the link to the Analyze costs page.

File: docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/static/create-cost-perspectives-24.png
[the remark this answers is not in the document]
Comment (charanya-j) [reply docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/static/create-cost-perspectives-24.png:1]: As I mentioned earlier, we are trying to restrict screenshots to the minimum. I don't think we need to add these screenshots as we are providing one from the default settings page. Moreover, this is not a new feature, we are enhancing an existing one.

File: docs/cloud-cost-management/3-use-ccm-cost-reporting/3-root-cost-analysis/analyze-cost-for-gcp-using-perspectives.md (hunk 2 above)
Comment (charanya-j) [reply docs/cloud-cost-management/3-use-ccm-cost-reporting/3-root-cost-analysis/analyze-cost-for-gcp-using-perspectives.md:69]: This one I'll add

File: docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (sdsingh07) [reply docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md:131]: But, they don't know GCP Invoice month is present in the analyze cost page

File: docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md (hunk 3 above)
Comment (sdsingh07) [reply docs/cloud-cost-management/3-use-ccm-cost-reporting/1-ccm-perspectives/perspective-preferences.md:19]: Yes, we can add it to the release notes.

**Pull request merged**